### PR TITLE
Reorganize testing

### PR DESCRIPTION
**Pull request by jonlawlor**
This is the first step to enable docker based testing for OpenBLAS.  The new code structure permits different installation and testing scripts for each OS & BLAS library combination, which is necessary for testing on docker images (because the commands will be different).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -30,25 +30,12 @@ matrix:
...
- # Required for coverage.
- - go get golang.org/x/tools/cmd/cover
- - go get github.com/mattn/goveralls
-
-# Install the appropriate blas library if we are using cgo.
+# Install the appropriate blas library (if any) and associated gonum software.
 install:
- - source .travis/$TRAVIS_OS_NAME/$BLAS_LIB/install.sh
- - go get github.com/gonum/floats
+ - source ${TRAVIS_BUILD_DIR}/.travis/$TRAVIS_OS_NAME/$BLAS_LIB/install.sh
```
Comment (kortschak) [inline .travis.yml]: Do these paths exist in this commit?

File: .travis.yml [hunk 2]
```diff
@@ -30,25 +30,12 @@ matrix:
...
 
-# Get deps, build, test, ensure the code is gofmt'ed and check that go generate was run.
-# Move into the native directory if we aren't using an external blas lib.
-# If we are building as gonum, then we have access to the coveralls api key, so we can run coverage as well.
 script:
- - if [[ "$BLAS_LIB" == "gonum" ]]; then pushd native; fi
- - go get -d -t -v ./...
- - go test -a -v ./...
- - go test -a -tags noasm -v ./...
+ - source ${TRAVIS_BUILD_DIR}/.travis/$TRAVIS_OS_NAME/$BLAS_LIB/test.sh
```
Comment (kortschak) [inline .travis.yml]: Ditto here.

File: .travis/linux/ATLAS/install.sh [hunk 3]
```diff
@@ -6,3 +6,6 @@ sudo apt-get update -qq && sudo apt-get install -qq libatlas-base-dev
 # fetch and install gonum/blas against ATLAS
 export CGO_LDFLAGS="-L/usr/lib -lblas"
 go get github.com/gonum/blas
+
+# run the OS common installation script
+source ${TRAVIS_BUILD_DIR}/.travis/$TRAVIS_OS_NAME/install.sh
```
Comment (kortschak) [inline .travis/linux/ATLAS/install.sh]: Same question for these lines.

**Comment by kortschak** [comment]
This all looks OK to me (you can ignore the questions above, I think I'm clear on how this is working now).

LGTM

**Comment by jonlawlor** [comment]
Do you think we should modify the check-generate.sh scripts in other repos to exclude .travis/ as well?

**Comment by kortschak** [comment]
Don't bother if it's not necessary.

**Pull request merged**